### ISSUE TITLE
Feature request- Add Compatibility with the Legion y540 and Hybrid mode

### ISSUE THREAD

**Issue by TDG2** [ViRb3/LenovoController#3]

**Comment by ViRb3** [comment]
I don't own this machine so unfortunately I can't help. Happy to take PRs though.

**Comment by usernameak** [comment]
btw: Hybrid mode requires reboot anyway, so technically you can simply switch it in BIOS on startup

**Comment by pgrohe** [comment]
I have a 2019 Y540 (i7 9750H, GTX1650,1080p 60hz panel) and LenovoController is working for me.

I've confirmed the following all works:

- Touchpad lock on/off
- Function lock on/off
- Battery charge (normal mode, conservation mode, I don't use rapid charge)

I've set my Power Mode to Quiet and haven't heard any fans but this isn't a very accurate way of confirming the feature worked. If anyone can suggest a better way of confirming I'll have a look. Is it only changing fan speeds & activation thresholds? Or CPU/GPU voltage, max clock, etc...?

Overdrive is greyed out because I don't think Y540 supports it. At least not my model with the basic 1080p, 60hz panel.

I haven't tried Always On USB.

Hybrid mode works for me by changing the BIOS option to Switchable Graphics.  Doesn't need either LenovoController or Lenovo Vantage AFAIK. I did notice that after a fresh install of Win10 (including up to date NVIDIA driver) the first time I opened the BIOS ... the graphics card option wasn't present. I rebooted the machine and then it was there.  Don't know why. There's some Google hits for other people having the same issue.  But this is unrelated to LenovoController.

So in summary ... the tool is working fine for me.  No errors, instability, etc... on my machine with the features I've used.

To the developers, thanks for making this. Really great, simple replacement for the official bloatware.
